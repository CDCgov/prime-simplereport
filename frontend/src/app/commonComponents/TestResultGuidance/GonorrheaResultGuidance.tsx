import React from "react";
import { Trans, useTranslation } from "react-i18next";

const GonorrheaResultGuidance = () => {
  const { t } = useTranslation();

  return (
    <>
      <p className="text-bold sr-guidance-heading">
        {t("testResult.gonorrheaNotes.h1")}
      </p>
      <p>{t("testResult.gonorrheaNotes.positive.p0")}</p>
      <Trans
        parent="p"
        i18nKey="testResult.gonorrheaNotes.positive.p1"
        components={[
          <a
            href={t("testResult.gonorrheaNotes.positive.treatmentLink")}
            target="_blank"
            rel="noopener noreferrer"
            key="gonorrhea-treatment-link"
          >
            gonorrhea treatment link
          </a>,
        ]}
      />
    </>
  );
};

export default GonorrheaResultGuidance;
