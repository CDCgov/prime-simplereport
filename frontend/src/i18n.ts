import i18n from "i18next";
import { initReactI18next } from "react-i18next";

import { en } from "./lang/en";
import { es } from "./lang/es";
import { eo } from "./lang/eo";

export const resources = { en, es, eo };

i18n.use(initReactI18next).init({
  resources,
  lng: localStorage.getItem("language") || "en",
  interpolation: {
    escapeValue: false,
  },
});

export default i18n;
