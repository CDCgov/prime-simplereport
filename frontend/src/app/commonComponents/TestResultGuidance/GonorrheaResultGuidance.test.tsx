import { render, screen } from "@testing-library/react";

import "../../../i18n";

import GonorrheaResultGuidance from "./GonorrheaResultGuidance";

describe("GonorrheaResultGuidance", () => {
  it("displays guidance for a Gonorrhea result", () => {
    const { container } = render(<GonorrheaResultGuidance />);
    expect(screen.getByText("For Gonorrhea:")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
