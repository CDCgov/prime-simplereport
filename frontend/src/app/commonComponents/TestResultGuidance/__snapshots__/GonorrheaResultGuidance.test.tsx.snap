// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GonorrheaResultGuidance displays guidance for a Gonorrhea result 1`] = `
<div>
  <p
    class="text-bold sr-guidance-heading"
  >
    For Gonorrhea:
  </p>
  <p>
    If you have a positive result, you will need a follow-up test to confirm your results. The organization that provided your test should be able to answer questions and provide referrals for follow-up testing.
  </p>
  <p>
    <a
      href="https://www.cdc.gov/gonorrhea/about/index.html"
      rel="noopener noreferrer"
      target="_blank"
    >
      Visit the CDC website to learn more about a positive Gonorrhea result.
    </a>
     (cdc.gov/gonorrhea/about).
  </p>
</div>
`;
